chaoscenter/web/src/utils/probes.ts

```diff
@@ -16,36 +24,139 @@ export function getProbeDetails(probe: ProbeAttributes): string[][] {
+  });
+  return probeDetails;
+}
+
 export function getProbeProperties(probe: ProbeAttributes): string[][] {
   const probeProperties: string[][] = [];
   Object.entries(destructure(probe.runProperties)).forEach(property => {
     const key = property[0];
-    const value =
-      probe.type === 'httpProbe' && property[0] === 'probeTimeout'
-        ? `${property[1]} ms`
-        : typeof property[1] != 'boolean' && property[0] !== 'retry'
-        ? `${property[1]} seconds`
-        : `${property[1].toString()}`;
+    const value = property[1].toString();
 
     probeProperties.push([key, value]);
   });
   return probeProperties;
 }
 
-// export function getIcon(type: ProbeType): IconName {
-//   switch (type) {
-//     case ProbeType.HTTP:
-//       return 'http-probe';
-//     case ProbeType.CMD:
-//       return 'cmd-probe';
-//     case ProbeType.PROM:
-//       return 'service-prometheus';
-//     case ProbeType.K8S:
-//       return 'app-kubernetes';
-//     default:
-//       return 'custom-approval';
-//   }
-// }
+export function getProbePropertiesFromParsedProbe(probe: Probe): string[][] {
+  const probeProperties: string[][] = [];
+
+  const handleProbeProperties = (property: [string, string]): void => {
+    // Probe properties include both properties and details
+    // This if checks only allows properties
+    if (
+      property[0] === 'probeTimeout' ||
+      property[0] === 'interval' ||
+      property[0] === 'attempt' ||
+      property[0] === 'evaluationTimeout' ||
+      property[0] === 'retry' ||
+      property[0] === 'probePollingInterval' ||
+      property[0] === 'initialDelay' ||
+      property[0] === 'stopOnFailure'
+    ) {
+      const key = property[0];
+      const value = property[1].toString();
+
+      probeProperties.push([key, value]);
+    }
+  };
+
+  if (probe.type === ProbeType.HTTP && probe.infrastructureType === InfrastructureType.KUBERNETES)
+    Object.entries(destructure(probe.kubernetesHTTPProperties)).forEach(property => handleProbeProperties(property));
+  if (probe.type === ProbeType.CMD && probe.infrastructureType === InfrastructureType.KUBERNETES)
+    Object.entries(destructure(probe.kubernetesCMDProperties)).forEach(property => handleProbeProperties(property));
+  if (probe.type === ProbeType.PROM)
+    Object.entries(destructure(probe.promProperties)).forEach(property => handleProbeProperties(property));
+  if (probe.type === ProbeType.K8S)
+    Object.entries(destructure(probe.k8sProperties)).forEach(property => handleProbeProperties(property));
+
+  return probeProperties;
+}
+
+export function getNormalizedProbeName(type: ProbeType): string {
+  switch (type) {
+    case ProbeType.HTTP:
+      return 'HTTP';
+    case ProbeType.CMD:
+      return 'Command';
+    case ProbeType.PROM:
+      return 'Prometheus';
+    case ProbeType.K8S:
+      return 'Kubernetes';
+  }
+}
+
+export function getIcon(type: ProbeType): IconName {
```

Comment:
Can we add the icons for K8S and PROM too if they are published in harnessio!

Reply:
Will need more discussion on adding it to OSS, this is for immediate solution.